SharpPcap/PosixTimeval.cs

```diff
@@ -24,35 +24,35 @@ namespace SharpPcap
     /// <summary> POSIX.4 timeval</summary>
     public class PosixTimeval : IComparable<PosixTimeval>
     {
-        private static readonly DateTime epochDateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
-
-        internal long microsecondsPerMillisecond = 1000;
+        private static readonly DateTime EpochDateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
 
         /// <value>
         /// Number of seconds in the timeval
         /// </value>
-        virtual public ulong Seconds
+        public ulong Seconds
         {
-            get;
-            set;
+            get => (ulong)Value;
+            set => Value = (Value % 1) + value;
         }
 
         /// <value>
         /// Number of microseconds in the timeval
         /// </value>
-        virtual public ulong MicroSeconds
+        public ulong MicroSeconds
         {
-            get;
-            set;
+            get => (ulong)((Value % 1) * 1e6M);
+            set => Value = Seconds + (value * 1e-6M);
         }
 
+        /// <summary>
+        /// Seconds since epoch
+        /// </summary>
+        public decimal Value { get; set; }
```

Comment:
@chmorgan I used decimal type since it's a 128bit floating number big enough to fit seconds and micro/nano seconds with no round-off errors.

Also it will simplify operators logic